### PR TITLE
chore upgrade to react 16 / enzyme 3

### LINKED ISSUE #638: v4.3.0 has introduced a breaking change to our unit tests
**Do you want to request a *feature* or report a *bug*?**

- [x] I found a bug
- [ ] I want to propose a feature

**What is the current behavior?**
Before release 4.3.0 we had a test that did:
```
    // Create a non-null file
    const fileContents = 'file contents'
    const file = new Blob([fileContents], { type: 'text/plain' })

    wrapper.find('.FileDrop__Dropzone').at(0).simulate('drop', { dataTransfer: { files: [file] } })
    expect(wrapper.state('files')).toEqual([file])
    expect(onFileDrop).toBeCalled()
```

This passed, but now we get:
```
  ● FileDrop › calls onFileDrop when a file is dropped

    expect(received).toEqual(expected)

    Expected value to equal:
      [{Symbol(impl): {"_buffer": {"data": [102, 105, 108, 101, 32, 99, 111, 110, 116, 101, 110, 116, 115], "type": "Buffer"}, "type": "text/plain", Symbol(wrapper): [Circular]}}]
    Received:
      []
```

Pinning back to v4.2.13 fixes our test

**If the current behavior is a bug, please provide the steps to reproduce.**

1.
1.
1.

**What is the expected behavior?**

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Operating System and react-dropzone version.**

### PR DESCRIPTION
**Pull request by tobilen**
closes #638

**What kind of change does this PR introduce?**

- [ ] bugfix
- [ ] feature
- [ ] refactoring / style
- [X] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [X] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [X] Not relevant

**Summary**

This PR updates react-dropzone to build with react 16 and be tested with enzyme 3. There is currently logic whose tests need to be refactored to work with enzyme 3, and this PR provides a baseline for this problem. See https://github.com/react-dropzone/react-dropzone/issues/638

**Does this PR introduce a breaking change?**
No breaking change

**Other information**

### CODE REVIEW

**Comment by tobilen** [comment]
as a side note, the `nested Dropzone component behavior` test block is kind of messy, with stuff leaking into each other deliberately. i tried to clean it up a but, but ideally we replace sinon with jest spies and reset them in between tests ( https://jestjs.io/docs/en/configuration.html#clearmocks-boolean )

**Comment by okonet** [comment]
@tobilen can you work on ditching the sinon please? this is something I'd like to do but no time yet.

File: testSetup.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
-/* eslint prefer-template: 0 */
+import Enzyme from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+
+Enzyme.configure({ adapter: new Adapter() })
 require('jest-enzyme')
```
Comment (okonet) [inline testSetup.js]: Use `import` here too and group with other imports?

File: README.md [hunk 2]
```diff
@@ -92,6 +92,37 @@ See https://react-dropzone.netlify.com/#proptypes
 
 *Important*: `react-dropzone` doesn't manage dropped files. You need to destroy the object URL yourself whenever you don't need the `preview` by calling `window.URL.revokeObjectURL(file.preview);` to avoid memory leaks.
 
+### Testing
+
+*Important*: for performance reasons, `react-dropzone` makes its drag'n'drop callbacks asynchronous. In order to properly test this, you may want to utilize a helper function to run all promises like this:
```
Comment (okonet) [inline README.md]: It's not performance reasons but plugins. See https://github.com/react-dropzone/react-dropzone/pull/616

**Comment by tobilen** [comment]
Can do, but i would do it in a different pr

**Comment by okonet** [comment]
I’m fine with a different PR! Thanks!

**Pull request merged**